views/js/runner/provider/qti.js

```diff
@@ -292,171 +293,183 @@ define([
+                        })
+                        .catch(function(err){
+                            self.trigger('error', err);
+                        });
+                })
+                .on('timeout', function(scope, ref){
 
-                context.isTimeout = true;
+                    var context = self.getTestContext();
 
-                self.disableItem(context.itemUri);
+                    context.isTimeout = true;
+
+                    self.disableItem(context.itemUri);
 
-                submit()
-                    .then(updateStats)
-                    .then(function() {
-                        self.trigger('alert', __('Time limit reached, this part of the test has ended.'), function() {
-                            computeNext('timeout', {
-                                scope: scope,
-                                ref: ref
+                    submit()
+                        .then(updateStats)
+                        .then(function() {
+                            self.trigger('alert', __('Time limit reached, this part of the test has ended.'), function() {
+                                computeNext('timeout', {
+                                    scope: scope,
+                                    ref: ref
+                                });
                             });
+                        })
+                        .catch(function(err){
+                            self.trigger('error', err);
                         });
-                    })
-                    .catch(function(err){
-                        self.trigger('error', err);
-                    });
-            })
-            .on('pause', function(){
-                // will notify the server that the test was auto paused
-                self.getProxy().callTestAction('pause').then(function() {
-                    self.trigger('leave', {
-                        code: self.getTestData().states.suspended
-                    });
                 })
-                .catch(function(err){
-                    self.trigger('error', err);
-                });
-            })
-            .on('renderitem', function(itemRef){
-
-                var context = this.getTestContext();
-                var states = this.getTestData().itemStates;
-                var warning = false;
-
-                /**
-                 * Get the label of the current item
-                 * @returns {String} the label (fallback to the item identifier);
-                 */
-                var getItemLabel = function getItemLabel(){
-                    var item = mapHelper.getItem(self.getTestMap(), context.itemIdentifier);
-                    return item && item.label ? item.label : context.itemIdentifier;
-                };
+                .on('pause', function(data){
+                    var pause;
 
-                this.trigger('enablenav enabletools');
+                    if (!self.getState('disconnected')) {
```

Comment:
just thinking, but should'n we check the "disconnected" state directly within the proxy ? for all requests ?